File: inventarioapi/src/main/java/inventario/inventarioapi/model/InventarioData.java
package inventario.inventarioapi.model;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "INVENTARIO_DATA")
public class InventarioData {
    @Id
    @Column(name = "CODIGO")
    private Long codigo;

    @ManyToOne
    @JoinColumn(name= "CODLOJA")
    private LojaFisica loja;

    @Column(name = "DATA")
    private LocalDateTime data;

    @Column(name = "HORA")
    private String hora;

    @Column(name = "NOME")
    private String nome;

    @Column(name= "DISPONIVEL_EFD")
    private int disponivelEfd;

    public InventarioData() {
    }

    public InventarioData(Long codigo, LojaFisica loja, LocalDateTime data, String hora, String nome, int disponivelEfd) {
        this.codigo = codigo;
        this.loja = loja;
        this.data = data;
        this.hora = hora;
        this.nome = nome;
        this.disponivelEfd = disponivelEfd;
    }

    public Long getCodigo() {
        return codigo;
    }

    public void setCodigo(Long codigo) {
        this.codigo = codigo;
    }

    public LojaFisica getLoja() {
        return loja;
    }

    public void setLoja(LojaFisica loja) {
        this.loja = loja;
    }

    public LocalDateTime getData() {
        return data;
    }

    public void setData(LocalDateTime data) {
        this.data = data;
    }

    public String getHora() {
        return hora;
    }

    public void setHora(String hora) {
        this.hora = hora;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getDisponivelEfd() {
        return disponivelEfd;
    }

    public void setDisponivelEfd(int disponivelEfd) {
        this.disponivelEfd = disponivelEfd;
    }

}

File: inventarioapp/lib/src/models/inventario_data.dart
import 'empresa.dart';

class InventarioData {
  int? _codigo;
  Empresa? _loja;
  DateTime? _data;
  String? _hora;
  String? _nome;
  int? _disponivelEfd;

  InventarioData(
    this._codigo,
    this._loja,
    this._data,
    this._hora,
    this._nome,
    this._disponivelEfd,
  );


  factory InventarioData.fromJson(Map<String, dynamic> json) {
    return InventarioData(
      json['codigo'] as int,
      json['loja'] != null
          ? Empresa.fromJson(json['loja'])
          : null as Empresa?,
      json['data'] != null ? DateTime.parse(json['data']) : null as DateTime?,
      json['hora'] as String?,
      json['nome'] as String?,
      json['disponivelEfd'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'codigo': _codigo,
      'loja': _loja?.toJson(),
      'data': _data?.toIso8601String(),
      'hora': _hora,
      'nome': _nome,
      'disponivelEfd': _disponivelEfd,
    };
  }

  int? get disponivelEfd => _disponivelEfd;

  set disponivelEfd(int? value) {
    _disponivelEfd = value;
  }

  String? get nome => _nome;

  set nome(String? value) {
    _nome = value;
  }

  String? get hora => _hora;

  set hora(String? value) {
    _hora = value;
  }

  DateTime? get data => _data;

  set data(DateTime? value) {
    _data = value;
  }

  Empresa? get loja => _loja;

  set loja(Empresa? value) {
    _loja = value;
  }

  int? get codigo => _codigo;

  set codigo(int? value) {
    _codigo = value;
  }
}
